src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,26 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import mate.academy.spring.service.dto.mapping.impl.response.UserResponseMapper;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final UserResponseMapper userResponseMapper;
+
+    public UserController(UserService userService, UserResponseMapper userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto findByEmail(@RequestParam String email) {
+        return userResponseMapper.toDto(userService.findByEmail(email).get());
+    }
```

Comment:
we can check here if we get null from Optional

Reply:
we can leave it as it is for now